tools/serverpod_cli/lib/src/analyzer/entities/validation/restrictions.dart

```diff
@@ -313,15 +322,58 @@ class Restrictions {
     }
 
     var def = documentDefinition;
-    if (def is ClassDefinition) {
-      var field = def.findField(parentNodeName);
+    if (def is! ClassDefinition) return errors;
 
-      if (field?.scalarFieldName != null && !type.endsWith('?')) {
-        errors.add(SourceSpanSeverityException(
-          'Fields with a protocol relations must be nullable (e.g. $parentNodeName: $type?).',
-          span,
-        ));
-      }
+    var field = def.findField(parentNodeName);
+    if (field == null) return errors;
+
+    if (field.scalarFieldName != null && !type.endsWith('?')) {
+      errors.add(SourceSpanSeverityException(
+        'Fields with a protocol relations must be nullable (e.g. $parentNodeName: $type?).',
+        span,
+      ));
+    }
+
+    String? parsedType = _extractReferenceClassName(field);
+
+    var classes = entityRelations?.classNames[parsedType];
+
+    if (field.hasRelationPointer && (classes == null || classes.isEmpty)) {
+      errors.add(SourceSpanSeverityException(
+        'The class "$parsedType" was not found in any protocol.',
+        span,
+      ));
+      return errors;
+    }
+
+    if (field.hasRelationPointer && !_hasTableDefined(classes)) {
+      errors.add(SourceSpanSeverityException(
+        'The class "$parsedType" must have a "table" property defined to be used in a relation.',
+        span,
+      ));
+    }
+
+    if (!(field.type.isList)) return errors;
+
+    var referenceClass = classes?.first;
+    if (referenceClass is! ClassDefinition) return errors;
```

Comment:
No it means the "class" is not a class but an enum, and hence is not able to have a relation.

With that said that does mean we have a problem. I think the code would generate the Error: 'The class "$parsedType" must have a "table" property defined to be used in a relation.' is this would happen, but we probably should add a test case for this and give an explicit error for this situation!

Reply:
I added a new case that will print `Only classes can be used in relations, "$parsedType" is not a class.` if a none class relation is detected.